src/GitHub.VisualStudio.UI/UI/Controls/PullRequestStatusCircle.xaml.cs

```diff
@@ -0,0 +1,297 @@
+                var angleDegrees = adjustedDegrees - 315;
+                var angleRadians = ToRadians(angleDegrees);
+                var tan = Math.Tan(angleRadians);
+                var oppositeEdge = tan * Radius;
+                return new[] { Origin, topMiddle, topRight, bottomRight, bottomLeft, new Point(bottomLeft.X, bottomLeft.Y - oppositeEdge) };
+            }
+
+            if (adjustedDegrees >= 0 && adjustedDegrees < 45)
+            {
+                var angleDegrees = adjustedDegrees;
+                var angleRadians = ToRadians(angleDegrees);
+                var tan = Math.Tan(angleRadians);
+                var oppositeEdge = tan * Radius;
+                return new[] { Origin, topMiddle, topRight, bottomRight, bottomLeft, new Point(bottomLeft.X, bottomLeft.Y - Radius - oppositeEdge) };
+            }
+
+            if (adjustedDegrees >= 45 && adjustedDegrees < 90)
+            {
+                var angleDegrees = adjustedDegrees - 45;
+                var angleRadians = ToRadians(angleDegrees);
+                var tan = Math.Tan(angleRadians);
+                var oppositeEdge = tan * Radius;
+                return new[] { Origin, topMiddle, topRight, bottomRight, bottomLeft, topLeft, new Point(topLeft.X + oppositeEdge, topLeft.Y) };
+            }
+
+            throw new InvalidOperationException();
+        }
+
+        public PullRequestStatusCircle()
+        {
+            InitializeComponent();
+            GeneratePolygons();
+            GenerateMask();
+        }
+
+        private void GeneratePolygons()
+        {
+            ErrorPolygon.Points = new PointCollection(GeneratePoints((float)ErrorCount / TotalCount));
+            SuccessPolygon.Points = new PointCollection(GeneratePoints((float)(SuccessCount + ErrorCount) / TotalCount));
+            PendingPolygon.Points = new PointCollection(GeneratePoints((float)(SuccessCount + ErrorCount + PendingCount) / TotalCount));
+        }
+
+        private void GenerateMask()
+        {
+            var pendingPolygonClip = new CombinedGeometry(
+                GeometryCombineMode.Exclude,
+                new EllipseGeometry(Origin, Radius, Radius),
+                new EllipseGeometry(Origin, InnerRadius, InnerRadius));
+
+            PendingPolygon.Clip = pendingPolygonClip;
+            SuccessPolygon.Clip = pendingPolygonClip;
+            ErrorPolygon.Clip = pendingPolygonClip;
+        }
+
+        private Point Origin => new Point(Radius + XAdjust, Radius + YAdjust);
+
+        private double Diameter => Radius * 2;
+
+        private double XAdjust => (ActualWidth - Diameter) / 2;
+
+        private double YAdjust => (ActualHeight - Diameter) / 2;
+
+        private int TotalCount => ErrorCount + SuccessCount + PendingCount;
+
+        public int ErrorCount
+        {
+            get => (int)GetValue(ErrorCountProperty);
+            set
+            {
+                SetValue(ErrorCountProperty, value);
+                GeneratePolygons();
```

Comment:
Dependency property CLR setter wrappers shouldn't do anything other than call `SetValue` because when they're set from XAML the CLR setter isn't actually called (XAML will call `SetValue` directly). Instead of doing this, you should be adding a property changed handler on dependency property registration. See [here](https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/custom-dependency-properties#implementing-the-wrapper) for more details.

Reply:
Fixed